src/reduce.py

```diff
@@ -12,8 +12,13 @@ def reduce(f: Callable[[A], B], x: list[A]) -> B:
     >>> reduce(lambda x,y: x+y, [1, 2, 3])
     6
     """
-    assert len(x) >= 2
-    ...  # FIXME
+    if len(x) <= 2:
+        return x[0]
+    x = iter(x)
+    result = f(next(x), next(x))
```

Comment:
Clever and a little cheeky. It works in Python, but be careful. Not all languages guarantee that function arguments are evaluated left-to-right.

Reply:
Ok! This scheme should be more general then, right?
```python
first_element = next(x),
second_element = next(x)
result = f(first_element, second_element)
```

Reply:
Yes. But as I said, the solution you have works in Python because we are guaranteed that the function arguments are evaluated that way. In C, for example, the compiler gets to choose, and then you couldn't be sure it would work, as the first element might go as the second argument and vice versa.